### ISSUE TITLE
🐛 emoji not supported in CSS parser

### ISSUE THREAD

**Issue by mayank99** [biomejs/biome#3627]
### Environment information

<details>

<summary>biome rage</summary>

```
CLI:
  Version:                      1.9.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.16.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/9.6.0"

Biome Configuration:
  Status:                       unset
```

</details>

### What happened?

When a custom property name contains an emoji, Biome fails to parse it.

```css
p {
  --✨-color: red;
  color: var(--✨-color);
}
```

I get errors in my IDE from `biome(parse)` in the form of "Expected a declaration, or an at rule but instead found '--✨'." and "Expected a declaration item but instead found '--✨'."

~~See [playground](https://biomejs.dev/playground/?files.main.css=cAAgAHsACgAgACAALQAtAD7YVN0tAGMAbwBsAG8AcgA6ACAAcgBlAGQAOwAKACAAIABjAG8AbABvAHIAOgAgAHYAYQByACgALQAtAD7YVN0tAGMAbwBsAG8AcgApADsACgB9AA%3D%3D).~~

See [updated playground](https://biomejs.dev/playground/?files.main.css=cAAgAHsACgAgACAALQAtAD7YVN0tAGMAbwBsAG8AcgAtADAAOgAgAHYAYQByACgALQAtAD3YG9wsACAAcgBlAGQAKQA7AAoAIAAgAAoAIAAgAC0ALQACJg%2F%2BLQBjAG8AbABvAHIALQAxADoAIAB2AGEAcgAoAC0ALQA%2B2FTdLQBjAG8AbABvAHIALQAwACkAOwAKACAAIAAKACAAIAAtAC0AKCctAGMAbwBsAG8AcgAtADIAOgAgAHYAYQByACgALQAtAAImD%2F5jAG8AbABvAHIALQAxACkAOwAKACAAIAAKACAAIABjAG8AbABvAHIAOgAgAHYAYQByACgALQAtACgnLQBjAG8AbABvAHIALQAyACkAOwAKAH0A) and [related comment](https://github.com/biomejs/biome/issues/3627#issuecomment-2392388022).

### Expected result

Emojis are allowed in custom property names, so Biome should be able to parse them.

![](https://github.com/user-attachments/assets/68e3fbd1-1bd6-481f-8e60-39d85346c483)

~~It's worth noting that since Biome doesn't support formatting/linting CSS, it should not even be parsing my CSS files.~~

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
> It's worth noting that since Biome doesn't support formatting/linting CSS, it should not even be parsing my CSS files.

That's intended by design, as Biome aims to function as a language server, so it needs to provide parsing diagnostics regardless.

**Comment by Conaclos** [comment]
Minimal code to reproduce the panic in debug mode:

```js
let {
    p = function () { let l; },
} = o;
```

**Comment by Tom-Aive** [comment]
This issue also applies to property names starting with a number.
`--500-grey` throws `Expected a declaration, or an at rule but instead found` while `--grey-500` doesn't.

I'm not sure the PR fixes this issue entirely.

Do you want me to open another issue ?

**Comment by Conaclos** [comment]
> I've looked at the PR and this specific issue doesn't seem to be fixed.

The PR fixed also this issue. You can test it on the [Playground](https://biomejs.dev/playground/?files.main.css=UAAgAHsACgAgACAAYwBvAGwAbwByADoAIAAtAC0ANQAwADAALQBnAHIAZQB5ADsACgB9AA%3D%3D).

**Comment by Tom-Aive** [comment]
Perfect, thanks for the great work. 🙏

**Comment by mayank99** [comment]
It looks like this wasn't completely fixed. It still breaks with certain emojis. See [playground](https://biomejs.dev/playground/?files.main.css=cAAgAHsACgAgACAALQAtAD7YVN0tAGMAbwBsAG8AcgAtADAAOgAgAHIAZQBkADsACgAgACAACgAgACAALQAtAAImD%2F4tAGMAbwBsAG8AcgAtADEAOgAgAHYAYQByACgALQAtAD7YVN0tAGMAbwBsAG8AcgAtADAAKQA7AAoAIAAgAAoAIAAgAC0ALQAoJy0AYwBvAGwAbwByAC0AMgA6ACAAdgBhAHIAKAAtAC0AAiYP%2FmMAbwBsAG8AcgAtADEAKQA7AAoAIAAgAAoAIAAgAGMAbwBsAG8AcgA6ACAAdgBhAHIAKAAtAC0AKCctAGMAbwBsAG8AcgAtADIAKQA7AAoAfQA%3D). Should this issue be reopened?

**Comment by Conaclos** [comment]
@mayank99 Actually they are emoji that are not allowed as identifiers according to the specification (See the [identifier diagram](https://drafts.csswg.org/css-syntax-3/#ident-token-diagram) and the specification of [non-ascii codepoint](https://drafts.csswg.org/css-syntax-3/#non-ascii-ident-code-point)). Even the current draft specification [doesn't include its emoji](https://drafts.csswg.org/css-syntax/#non-ascii-ident-code-point).

- `✨` is the codepoint `U+2728`
- `☂` is the codepoint `U+2602`

**Comment by mayank99** [comment]
Yeah, that's interesting because it works in all three browsers.

![](https://github.com/user-attachments/assets/05118de8-70b8-4019-a78b-b7cfb8591297)

Since browsers already work this way, I wonder if the spec needs to be updated. This could be a good issue to raise with the CSSWG.

**Comment by Conaclos** [comment]
We should take a look at some browser implementations and indeed raise the issue.

**Comment by Conaclos** [comment]
I raised the issue on to the folks in charge of CSS specification. See https://github.com/w3c/csswg-drafts/issues/11005

**Comment by mayank99** [comment]
@Conaclos Thanks so much for raising the issue with CSSWG and for accepting the additional emoji in Biome as a stopgap. I tried out in 1.9.4 just now, and it works great! ✨
